### PR TITLE
add: refactored signing out the user from the application

### LINKED ISSUE #143: Update Logout function on frontend to work with endpoint
Logout function in store should also call proper endpoint (#140 ) to delete cookie.

### PR DESCRIPTION
**Pull request by kpagacz**
* wipes the user data
* wipes the session cookie
* improved docs a little bit
Closes #143

### CODE REVIEW

File: frontend/src/store.js [hunk 1]
```diff
@@ -12,9 +13,18 @@ const store = createStore({
...
     mutations: {
         /**
-        * This will clear all user's data and unauthorize current session
+        * Signs user out of the application session.
+        *
+        * Wipes the stored user data and the session cookie.
+        *
+        * @throws {Error} if wiping out the session cookie was not successful
         */
         logOut(state) {
```
Comment (karolkoz) [inline frontend/src/store.js:22]: This should be async function

File: frontend/src/httpLayers/signout.http.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
+const { default: wofApi } = require("./wofApi");
```
Comment (karolkoz) [inline frontend/src/httpLayers/signout.http.js:1]: It is unnecessary. Below we have it imported

**Review by karolkoz — CHANGES_REQUESTED** [review]

File: frontend/src/httpLayers/signout.http.js (hunk 2 above)
Comment (kpagacz) [reply frontend/src/httpLayers/signout.http.js:1]: Mhm, I must have missed one XD

Comment (kpagacz) [reply frontend/src/httpLayers/signout.http.js:1]: Removed

File: frontend/src/store.js (hunk 1 above)
Comment (kpagacz) [reply frontend/src/store.js:22]: changed the modifier

**Comment by kpagacz** [comment]
This is ready for re-review

File: frontend/src/store.js [hunk 3]
```diff
@@ -12,9 +13,18 @@ const store = createStore({
...
         /**
-        * This will clear all user's data and unauthorize current session
+        * Signs user out of the application session.
+        *
+        * Wipes the stored user data and the session cookie.
+        *
+        * @throws {Error} if wiping out the session cookie was not successful
         */
-        logOut(state) {
+        async logOut(state) {
```
Comment (karolkozlo) [inline frontend/src/store.js:22]: If this is async, please move it to actions

Comment (karolkozlo) [reply frontend/src/store.js:22]: And remember to update use of this in WofHeader component.
(i mean, use 'dispatch' instead of 'commit')

Comment (kpagacz) [reply frontend/src/store.js:22]: These vuex intricacies elude me unfortunately :( But I will read up on them

Comment (kpagacz) [reply frontend/src/store.js]: Done

**Comment by kpagacz** [comment]
Ready for re-review

File: frontend/src/store.js [hunk 4]
```diff
@@ -1,76 +1,86 @@
...
-            }
-        }
+    /**
+     * Signs user out of the application session.
+     *
+     * Wipes the stored user data and the session cookie.
+     *
+     * @throws {Error} if wiping out the session cookie was not successful
+     */
+    async logOut(state) {
```
Comment (karolkozlo) [inline frontend/src/store.js:46]: Remove 'state' argument from here. If you want get access to state in action, you can do it with
```this.state.something = whatever;```

Comment (kpagacz) [reply frontend/src/store.js]: Done

**Review by karolkozlo — APPROVED** [review]

**Pull request merged**